grapher/draw.js

```diff
@@ -53,19 +53,32 @@ function drawGridRenderer() {
     rctx.strokeStyle = '#00C49A';
     rctx.lineWidth = 2.25;
 
-    for (const funcString of functionsToPlot) {
+    for (const func of functionsToPlot) {
         rctx.beginPath();
 
+        let lastRealX = -roffsetX / 2;
+
         // Plot the function points within the canvas bounds
         for (let x = 0; x < rcanvasWidth; x++) {
-            const realX = (x - rcanvasWidth / 2 + roffsetX) * 0.025;
-            const realY = -evaluateFunction(funcString, realX) * 40 + (rcanvasHeight / 2 - roffsetY);
+            const realX = getRealX(x);
+
+            let isAsymptote = false;
+            for (const asymptote of func["asymptotes"]) {
+                if (lastRealX < asymptote && realX > asymptote) {
```

Comment:
lmao this is what i was trying to do earlier but this is a much better implementation lol

Reply:
i need to extract this into its own function lmao